### PR TITLE
Separated messaging layer from UserChannel

### PR DESCRIPTION
**Pull request by fhubi**
- Created a new Messaging.Abstractions project for the abstractions.
- Created IMessagingService interface and the MessageRouter implements this interface.
- UserChannel and the Fdc3DesktopAgentMessageRouterService use the new abstractions project and interfaces.
- Updated tests

### CODE REVIEW

File: src/messaging/dotnet/src/Core/MessageBufferJsonExtensions.cs
Comment (ZKRobi) [inline src/messaging/dotnet/src/Core/MessageBufferJsonExtensions.cs:1]: I'd keep these as an extension as it's derived behavior so it's unnecessary to reimplement with the interface.

File: src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/DependencyInjection/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -34,6 +35,7 @@ public static class ServiceCollectionExtensions
             builderAction(builder);
         }
 
+        serviceCollection.AddSingleton<IMessagingService>(provider => provider.GetRequiredService<IMessageRouter>());
```
Comment (ZKRobi) [inline src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/DependencyInjection/ServiceCollectionExtensions.cs:38]: As discussed offline adding this registration together with the IMessageRouter registration is worth considering

File: src/messaging/dotnet/src/Core/PublishOptions.cs
[the remark this answers is not in the document]
Comment (fhubi) [reply src/messaging/dotnet/src/Core/PublishOptions.cs:1]: Removed the unnecessary interfaces.

File: src/messaging/dotnet/src/Abstractions/IMessageContext.cs
[the remark this answers is not in the document]
Comment (fhubi) [reply src/messaging/dotnet/src/Abstractions/IMessageContext.cs:1]: Removed the unnecessary interfaces.

File: src/messaging/dotnet/src/Core/MessageBufferJsonExtensions.cs
Comment (fhubi) [reply src/messaging/dotnet/src/Core/MessageBufferJsonExtensions.cs:1]: Added back MessageRouterJsonExtensions.

File: src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/DependencyInjection/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (fhubi) [reply src/fdc3/dotnet/DesktopAgent/src/MorganStanley.ComposeUI.DesktopAgent/DependencyInjection/ServiceCollectionExtensions.cs]: Done.

**Review by ZKRobi — APPROVED** [review]
Merge with squashing please :)

**Pull request merged**